### PR TITLE
Recommend Number.isNaN(), not global isNan()

### LINKED ISSUE #3026: Issue with "A re-introduction to JavaScript (JS tutorial)": Should use Number.isNaN rather than regular isNaN
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript

#### What information was incorrect, unhelpful, or incomplete?

In the article (Under Number) it says that "You can test for NaN using the built-in isNaN() function:"
However, this fuction also returns true on lots of other inputs, other than NaN
```
isNaN('hello') // true
isNaN(function(){}) // true
isNaN({hello: 'world'}) // true
```
But other than that, it is inconsistent as well:
```
isNaN([1]) // false
isNaN([1,2]) // true
isNaN('1') // false
```
#### Specific section or headline?
under Numbers / You can test for NaN using the built-in isNaN() function:

#### What did you expect to see?
Instead of isNaN, Number.isNaN produces the expected result:
```
Number.isNaN('hello') // false
Number.isNaN(function(){}) // false
Number.isNaN({hello: 'world'}) // false
Number.isNaN([1,2]) // false
```

Browser support is a little bit worse for Number.isNaN when compared to isNaN but it's supported in all the current browsers, plus that if you actually need it, you can always use a polyfill, as seen on
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#polyfill

#### Did you test this? If so, how?
Using Chrome DevTools

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/a_re-introduction_to_javascript`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/a_re-introduction_to_javascript/index.html
* Last commit: https://github.com/mdn/content/commit/3b80812a1d4f68e57f35fca763f50bf3e5dfe997
* Document last modified: 2021-02-19T19:57:02.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/3026

### CODE REVIEW

**Comment by Elchi3** [comment]
Should we also link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#description if the reader wants to understand why things are like this?

**Comment by sideshowbarker** [comment]
> Should we also link to [developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#description](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#description) if the reader wants to understand why things are like this?

Thanks, yep — added that link, as well as a link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#confusing_special-case_behavior

File: files/en-us/web/javascript/a_re-introduction_to_javascript/index.html [hunk 1]
```diff
@@ -119,9 +119,25 @@ <h2 id="Numbers">Numbers</h2>
 <pre class="brush: js">NaN + 5; // NaN
 </pre>
 
-<p>You can test for <code>NaN</code> using the built-in {{jsxref("Global_Objects/isNaN", "isNaN()")}} function:</p>
+<p>You can reliably test for <code>NaN</code> using the built-in {{jsxref("Number.isNaN", "Number.isNaN()")}} function, <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#description">which behaves just as its name implies</a>:</p>
```
Comment (Elchi3) [inline files/en-us/web/javascript/a_re-introduction_to_javascript/index.html:122]:
```suggestion
<p>You can reliably test for <code>NaN</code> using the built-in {{jsxref("Number.isNaN", "Number.isNaN()")}} function, <a href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#description">which behaves just as its name implies</a>:</p>
```

File: files/en-us/web/javascript/a_re-introduction_to_javascript/index.html [hunk 2]
```diff
@@ -119,9 +119,25 @@ <h2 id="Numbers">Numbers</h2>
...
+<pre class="brush: js">Number.isNaN(NaN); // true
+Number.isNaN('hello'); // false
+Number.isNaN('1'); // false
+Number.isNaN(undefined); // false
+Number.isNaN({}); // false
+Number.isNaN([1]) // false
+Number.isNaN([1,2]) // false
+</pre>
+
+<p>But don’t test for <code>NaN</code> using the global {{jsxref("Global_Objects/isNaN", "isNaN()")}} function, <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#confusing_special-case_behavior">which has unintuitive behavior</a>:</p>
```
Comment (Elchi3) [inline files/en-us/web/javascript/a_re-introduction_to_javascript/index.html:133]:
```suggestion
<p>But don’t test for <code>NaN</code> using the global {{jsxref("Global_Objects/isNaN", "isNaN()")}} function, <a href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#confusing_special-case_behavior">which has unintuitive behavior</a>:</p>
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Thanks! My understanding is that we use internal links without host. Not sure if that has changed, though.

**Comment by sideshowbarker** [comment]
> Thanks! My understanding is that we use internal links without host. Not sure if that has changed, though.

Not changed :) Thanks for catching that

**Review by Elchi3 — APPROVED** [review]
Cool! :)

**Comment by nathnolt** [comment]
Thanks 👍

**Comment by sideshowbarker** [comment]
@nathnolt Thanks for having raised the issue — it’s nice to be able to get this stuff clarified at point of use where readers/developers are going to first run into it

**Pull request merged**